### PR TITLE
intel-graphics-compiler: deprecate phases and replace it with runComm…

### PR DESCRIPTION
**Pull request by Stunkymonkey**
…andLocal

###### Motivation for this change
#28910

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/intel-graphics-compiler/default.nix [hunk 1]
```diff
@@ -1,6 +1,8 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchFromGitHub
 , cmake
+, runCommandLocal
 , pkg-config
```
Comment (nrdxp) [inline pkgs/development/compilers/intel-graphics-compiler/default.nix:6]: `pkg-config` isn't being used anywhere and can probably be removed now, unless it's actually needed.

File: pkgs/development/compilers/intel-graphics-compiler/default.nix [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
```
Comment (nrdxp) [inline pkgs/development/compilers/intel-graphics-compiler/default.nix:7]: what's the deal with the blank line? Maybe a nixpkgs-fmt is in order?

**Review by nrdxp — CHANGES_REQUESTED** [review]
very minor changes needed, but overall looks good and builds fine :+1:

File: pkgs/development/compilers/intel-graphics-compiler/default.nix (hunk 2 above)
Comment (Stunkymonkey) [reply pkgs/development/compilers/intel-graphics-compiler/default.nix]: was there before, but is now removed

File: pkgs/development/compilers/intel-graphics-compiler/default.nix (hunk 1 above)
Comment (Stunkymonkey) [reply pkgs/development/compilers/intel-graphics-compiler/default.nix]: sure. lets remove it

**Review by nrdxp — APPROVED** [review]

**Comment by nrdxp** [comment]
Result of `nixpkgs-review pr 133191` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>intel-compute-runtime</li>
    <li>intel-graphics-compiler</li>
  </ul>
</details>

**Pull request merged**